dynamic_programming/longest_common_subsequence.cpp

```diff
@@ -0,0 +1,205 @@
+ * @param string2 String 2
+ * @prints one of the Longest Common Subsequence
+ * @returns void
+ */
+void allPossibleLCS(int **c, char *a, int index, int i, int j, char *string1, char *string2)
+{
+	// If either i is 0 or j is 0  then there will be no LCS
+	while(i > 0 && j > 0)
+	{
+		// Comparing chars at index j-1 of string2 and index i-1 of string1 because indexing of strings starts from 0
+		// If Matched
+		if(string2[j-1] == string1[i-1])
+		{
+			a[index] = string2[j-1];
+			index++;
+			i--;
+			j--;
+		}
+		// If not matched
+		else
+		{	
+			if(c[i-1][j] > c[i][j-1])
+			{
+				i--;
+			}
+			else if(c[i-1][j] == c[i][j-1])
+			{
+                allPossibleLCS(c, a, index, i, j-1, string1, string2);
+				i--;
+			}
+			else
+			{
+				j--;
+			}
+		}
+	}
+    for(int i=index-1; i>=0; i--)
+	{
+		printf("%c", a[i]);
+	}
+	printf("\n\n");
+	return;
+}
+
+/**
+ * @breif max will find the maximum between 2 values
+ * @param a integer a
+ * @param b integer b
+ * @returns maximum value
+ */
+int max(int a, int b)
+{
+	if(a > b)
+	{
+		return a;
+	}
+	return b;
+}
+
+/**
+ * @brief LCS will make 2D-array with optimal solutions of sub-problems
+ * @param string1 String 1
+ * @param m Length of String 1
+ * @param string2 String 2
+ * @param n Length of String 2
+ * @returns 2D-array
+ */
+int** LCS(char *string1,int m, char *string2, int n)
+{
+	// Creating a 2D-array to store the solutions of all sub-problems
+	int **c;
+	
+	// Dynamically allocating memory for 2D-array
+	c = (int **)calloc(m+1, sizeof(int *));
+	for(int i=0; i<=m; i++)
+	{
+		*(c + i) = (int *)calloc(n+1, sizeof(int));
+	}
+	// c[i][j] will have the length of Longest common subsequence when length of string 1 is i and length of string 2 is j
+	
+	// If length of string 2 is 0 then there will be no Longest common subsequence
+	for(int i=1; i<=m; i++)
+	{
+		c[i][0] = 0;
+	}
+	
+	// If length of string 1 is 0 then there will be no Longest common subsequence
+	for(int j=0; j <=n; j++)
+	{
+		c[0][j] = 0;
+	}
+	
+	for(int i=0; i<m; i++)
+	{
+		for(int j=0; j<n; j++)
+		{
+			// If string1[i] == string2[j]
+			if(string1[i] == string2[j])
+			{
+				c[i+1][j+1] = c[i][j]+1;
+			}
+			// If string1[i] != string2[j]
+			else
+			{
+				c[i+1][j+1] = max(c[i][j+1], c[i+1][j]);
+			}
+		}
+	}
+	return c;
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+	char *string1, *string2;                      ///< 2 strings of max size MAX 
+	string1 = (char *)calloc(MAX, sizeof(char));  ///< memory allocation for string 1
+	string2 = (char *)calloc(MAX, sizeof(char));  ///< memory allocation for string 2
+	
+	/// Assigning Values of 'GAC' for string1
+	string1[0] = 'G';
+	string1[1] = 'A';
+	string1[2] = 'C';
+
+	/// Assigning Values of 'AGCAT' for string2
+	string2[0] = 'A';
+	string2[1] = 'G';
+	string2[2] = 'C';
+	string2[3] = 'A';
+	string2[4] = 'T';
+	
+	// Lengths of two Strings;
+	int lenOfStr1 = strlen(string1);
+	int lenOfStr2 = strlen(string2);
```

Comment:
Add a one-line description of the variables (see the line 167, 168, and 169).

Reply:
Updated all the Changes Requested Please review @Panquesito7 as soon as you can.
Thank You!